### DISCUSSION TITLE
Transactions with QueryService in Custom Resolver

### DISCUSSION THREAD

**Discussion by danjirovec** [TriPSs/nestjs-query#234] [Q&A] [answered]
Hello,
is there a way to execute the following mutation as a transaction when using TypeORM?

```
@Resolver(() => CustomerDto)
export class CustomerResolver {
  constructor(
    @InjectQueryService(Customer)
    readonly customersService: QueryService<CustomerDto>,
    @InjectQueryService(Membership)
    readonly membershipsService: QueryService<MembershipDto>,
  ) {}

  @Mutation(() => CustomerDto)
  @UseGuards(AuthGuard)
  async createCustomerWithMembership(@Args('input') input: CreateCustomerDto) {
    const [customer] = await this.customersService.query({
      filter: { email: { eq: input.email } },
      paging: { limit: 1 },
    });

    if (customer) {
      throw new ConflictException('Customer with this email already exists');
    }

    try {
      if (input.membership) {
        const membership = input.membership;
        const createdMembership =
          await this.membershipsService.createOne(membership);
        input.membership = createdMembership;
      }
      return await this.customersService.createOne(input);
    } catch (error) {
      throw error;
    }
  }
}
```

Tried using dataSource and entityManager but for whatever reason the transaction doesn't work. If I omit the check for existence of customer email and error gets thrown, the membership still gets created.

**Comment by TriPSs** [comment, marked as answer]
This is not possible by using the query services, you would need to create a transactions with TypeORM itself and use that query runner with the query builder (or the manager) and directly insert it yourself.

Since the `createOne` logic uses TypeORM `.save` you could also try to pass it all down to the customers service `createOne`, I think if configured correctly TypeORM will create/save the relation correctly.
